internal/engine/global_yaml_buildcontroller.go

```diff
@@ -34,23 +34,23 @@ func (c *GlobalYAMLBuildController) OnChange(ctx context.Context, st store.RStor
 
 	newK8sEntities := []k8s.K8sEntity{}
 	if m.K8sYAML() != c.lastGlobalYAMLManifest.K8sYAML() {
+		c.lastGlobalYAMLManifest = m
+		st.Dispatch(GlobalYAMLApplyStartedAction{})
+
 		entities, err := k8s.ParseYAMLFromString(m.K8sYAML())
 		if err != nil {
 			err = errors.Wrap(err, "Error parsing global_yaml")
 			logger.Get(ctx).Infof(err.Error())
-			c.lastGlobalYAMLManifest = m
-			st.Dispatch(GlobalYAMLApplyError{Error: err})
+			st.Dispatch(GlobalYAMLApplyCompleteAction{Error: err})
```

Comment:
I'm not wild about the fact that a dev will have to remember to dispatch a `ApplyComplete` action every time they put a new return statement in this code... maybe do this in a defer?

Reply:
It's not a defer, but I think this looks a whole lot better now. Thanks!